pfcpiface/notifier.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022-present Open Networking Foundation
+
+package pfcpiface
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	defaultNotificationInterval = 5 * time.Second
+)
+
+// downlinkDataNotifier
+type downlinkDataNotifier struct {
+	notifyChan chan<- uint64
+
+	notificationInterval time.Duration
+
+	// state keeps track of F-SEIDs and corresponding notification time in future
+	state sync.Map
+}
+
+func NewDownlinkDataNotifier(notifyChan chan<- uint64) *downlinkDataNotifier {
+	return &downlinkDataNotifier{
+		notifyChan:           notifyChan,
+		notificationInterval: defaultNotificationInterval,
+	}
+}
+
+func (n *downlinkDataNotifier) SetNotificationInterval(interval time.Duration) {
```

Comment:
All calls to this right now are immediately after creating the `downlinkDataNotifier`. Should this be an arg of `NewDownlinkDataNotifier`?